crates/snp_validation/src/config.rs

```diff
@@ -0,0 +1,208 @@
+    pub policy_flags: Option<u8>,
+
+    #[serde(default)]
+    #[serde(deserialize_with = "from_platform_string")]
+    pub platform_info_flags: Option<PlatformInfoFlags>,
+}
+
+/// This is a work-around for the `semver` crate which requires a patch value. This isn't
+/// applicable here, since the AMD attestation report Policy does not specify this.
+fn from_version_string<'de, D>(deserializer: D) -> Result<Option<Version>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: &str = Deserialize::deserialize(deserializer)?;
+
+    let version = match s.split('.').count() {
+        3 => Version::from_str(s).map_err(|_| Error::custom("invalid version string"))?,
+        2 => {
+            let new_version = format!("{}.0", s);
+            Version::from_str(&new_version).map_err(|_| Error::custom("invalid version string"))?
+        }
+        _ => {
+            return Err(Error::custom("invalid version string"));
+        }
+    };
+
+    Ok(Some(version))
+}
+
+fn from_hex<'de, D>(deserializer: D) -> Result<Option<Vec<Vec<u8>>>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: Vec<&str> = Deserialize::deserialize(deserializer)?;
+
+    let mut outer_vec = Vec::new();
+    for hash_string in s {
+        outer_vec.push(hex::decode(hash_string).map_err(|_| Error::custom("invalid hex"))?);
+    }
+
+    Ok(Some(outer_vec))
+}
+
+fn from_policy_string<'de, D>(deserializer: D) -> Result<Option<u8>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: &str = Deserialize::deserialize(deserializer)?;
+
+    let mut flags = FlagSet::<PolicyFlags>::from(PolicyFlags::Reserved);
+
+    for flag in s.to_string().split('|') {
+        match flag.trim() {
+            "Debug" => {
+                flags |= PolicyFlags::Debug;
+            }
+            "MigrateMA" => {
+                return Err(D::Error::custom("SNP migration not supported"));
+            }
+            "SingleSocket" => {
+                flags |= PolicyFlags::SingleSocket;
+            }
+            "SMT" => {
+                flags |= PolicyFlags::SMT;
+            }
+            _ => return Err(D::Error::custom(format!("unknown policy flag '{}'", flag))),
+        }
+    }
+
+    Ok(Some(flags.bits()))
+}
+
+fn from_platform_string<'de, D>(deserializer: D) -> Result<Option<PlatformInfoFlags>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: &str = Deserialize::deserialize(deserializer)?;
+
+    let flag = match s.trim() {
+        "SME" => PlatformInfoFlags::SME,
+        "TSME" => PlatformInfoFlags::TSME,
+        _ => {
+            return Err(D::Error::custom(format!(
+                "unknown platform info flag '{}'",
+                s
+            )))
+        }
+    };
+
+    Ok(Some(flag))
+}
+
+impl From<Config> for Version {
+    fn from(value: Config) -> Self {
+        match value.minimum_abi {
+            Some(v) => v,
+            None => Version::new(0, 0, 0),
+        }
+    }
+}
```

Comment:
Why not `From<Config> for Option<Version>` instead? Do we really want to silently ignore the absence of `minimum_abi`?

Reply:
If it's not specified, have it at zero, so it can still be compared against at validation time.

Reply:
Removed, no longer needed